omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptProposalController2_0Test.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs2_0;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.Concept;
+import org.openmrs.ConceptName;
+import org.openmrs.ConceptProposal;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RestTestConstants1_8;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.openmrs.util.OpenmrsConstants;
+
+import java.util.Collection;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class ConceptProposalController2_0Test extends MainResourceControllerTest {
+
+	private static ConceptProposal proposal;
+
+	@Before
+	public void before() throws Exception {
+		proposal = new ConceptProposal();
+		proposal.setOriginalText("original text");
+		proposal.setFinalText("final text");
+		proposal.setState(OpenmrsConstants.CONCEPT_PROPOSAL_UNMAPPED);
+		proposal.setComments("comments");
+		proposal.setMappedConcept(Context.getConceptService().getConceptByUuid(RestTestConstants1_8.CONCEPT_UUID));
+		Context.getConceptService().saveConceptProposal(proposal);
+	}
+
+	@Override
+	public String getURI() {
+		return "conceptproposal";
+	}
+
+	@Override
+	public String getUuid() {
+		return proposal.getUuid();
+	}
+
+	@Override
+	public long getAllCount() {
+		return 2;
+	}
+
+	@Test
+	public void shouldIgnoreProposal() throws Exception {
+		String json = "{\"action\": \"ignore\"}";
+
+		handle(newPostRequest(getURI() + "/" + proposal.getUuid(), json));
+
+		ConceptProposal ignoredProposal = Context.getConceptService().getConceptProposalByUuid(proposal.getUuid());
+		assertEquals(OpenmrsConstants.CONCEPT_PROPOSAL_REJECT, ignoredProposal.getState());
```

Comment:
How do we tell that the proposal was not the same before we ignored?

Reply:
https://github.com/openmrs/openmrs-module-webservices.rest/pull/494/commits/4ee35cc43b865fbb5bd786ae9d0237e17d3d3e4b